client/knowledge_base_search/src/containers/ResultsContainer/FullResult.js

```diff
@@ -38,51 +35,66 @@ class FullResult extends Component {
       id
     } = this.props;
 
+    let title_buffer = question !== undefined ? question.title : '';
+    let subtitle_buffer = question !== undefined ? question.subtitle : '';
+
     return (
-      <Element name={'scroll_to_answer_' + id} className='full_result_row--div'>
-        <h4>
-          Discovery answers…
-        </h4>
-        <h5 className='base--h5'>
-          Original post title
-        </h5>
-        <h2>
-          { title }
-        </h2>
-        <div className='full_result_container--div'>
-          <QuestionColumn
-            subtitle={subtitle}
-            userId={authorUserId}
-            userName={authorUsername}
-          />
-          <AnswerColumn
-            answer={answer}
-            userId={userId}
-            userName={username}
-            views={views}
-            upVotes={upModVotes}
-            downVotes={downModVotes}
-            accepted={accepted}
-          />
-        </div>
-        <div className='full_result_link--div'>
-          <div>
-            <Icon type='link' size='small' />
+      <div>
+      { title_buffer !== '' ? (
+        <Element name={'scroll_to_answer_' + id} className='full_result_row--div'>
+          <h4>
+            Discovery answers…
+          </h4>
+          <h5 className='base--h5'>
+            Original post title
+          </h5>
+          <h2>
+            { title_buffer }
+          </h2>
+          <div className='full_result_container--div'>
+            <QuestionColumn
+              subtitle={subtitle_buffer}
+              userId={authorUserId}
+              userName={authorUsername}
+            />
+            <AnswerColumn
+              answer={answer}
+              userId={userId}
+              userName={username}
+              views={views}
+              upVotes={upModVotes}
+              downVotes={downModVotes}
+              accepted={accepted}
+            />
           </div>
-          <div>
-            <h5>
-              View original full thread on Stack Exchange:
-            </h5>
-            <a
-              href={this.generateQuestionUrl(id)}
-              target='_blank'
-              className='base--a'
-            >
-              { this.generateQuestionUrl(id) }
-            </a>
+          <div className='full_result_link--div'>
+            <div>
+              <Icon type='link' size='small' />
+            </div>
+            <div>
+              <h5>
+                View original full thread on Stack Exchange:
+              </h5>
+              <a
+                href={this.generateQuestionUrl(id)}
+                target='_blank'
+                className='base--a'
+              >
+                { this.generateQuestionUrl(id) }
+              </a>
+            </div>
           </div>
-        </div>
-      </Element>
+        </Element>
+      ) : (
+        <Element name='scroll_to_answer_undefined' className='results_loading_icon'>
```

Comment:
Nope, this is a way to catch the dynamic element that is created when a document hasn't been fetched yet, but is being fetched. The loading spinner is there in the mean time.